### ISSUE TITLE
numerical evaluation of li[2]

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1691]
Imported from SourceForge on 2024-07-04 23:17:47
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2019-09-26 11:08:18
Original: https://sourceforge.net/p/maxima/bugs/3582

---

Using Maxima 5.42.2 and SBCL 1.4.2, I get
~~~
(%i1)	float(rectform((3/4) *exp(2*%pi * %i *  25 / 180 )));
(%o1)	0.5745333323392335*%i+0.4820907072649045
(%i2)	li[2](%);
Maxima encountered a Lisp error:
 Control stack exhausted (no more space for function call frames).
~~~

Cha nge float to bfloat, the value is OK

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 23:17:48
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2019-09-27 13:31:25
Original: https://sourceforge.net/p/maxima/bugs/3582/#cbec

---

The function bigfloat::li2numer uses  the transformations that relate the value of li[2]  at x to the value at 1/x and the value at x to the value at /(1-x).  For the input
~~~
0.5745333323392335*%i+0.4820907072649045
~~~
these transformations yield the loop:  use 1/(1-x) transformation, use 1/x transformation, and infinitely repeat.

Further, for the recalcitrant point exp(%i %pi / 3), all these transformations map the point back onto the unit circle.  The Macularin series for li[2] converges on the entire unit circle, but its converagence rate is slow.  For an efficient numerical evaluation, I'd gusess there needs to be another series that has faster convergence near the unit circle.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 23:17:52
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2020-04-26 19:04:24
Original: https://sourceforge.net/p/maxima/bugs/3582/#86a2

---

- **status**: open --> closed
- **assigned_to**: Dan Gildea

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 23:17:55
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2020-04-26 19:04:25
Original: https://sourceforge.net/p/maxima/bugs/3582/#d3d9

---

Fixed by [7fe723476cafade7c4babf5a44ef98708e2ab952]
